### PR TITLE
feat: mt pelerin fiat ramp

### PR DESCRIPTION
**Pull request by stackedq**
## Description

on top of https://github.com/shapeshift/lib/pull/986
adds Mt Pelerin fiat ramp under a new feature flag.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

https://app.dework.xyz/shapeshift-dao/product-bounties?taskId=1157933e-3ca4-4e24-a326-17da71d87a6f

## Risk

none

## Testing

enable `MtPelerinFiatRamp` flag. there should be a new fiat ramp in the buy/sell modal.

## Screenshots (if applicable)

Screen Shot 2022-08-23 at 15 40 52
Screen Shot 2022-08-23 at 15 41 05

### CODE REVIEW

**Comment by stackedq** [comment]
converting to draft until lib package is merged.

File: src/components/Modals/FiatRamps/views/RampsList.tsx [hunk 1]
```diff
@@ -24,13 +26,16 @@ type RampsListProps = {
 
 export const RampsList: React.FC<RampsListProps> = ({ setFiatRampProvider }) => {
   const history = useHistory()
+  const featureFlags = useAppSelector(selectFeatureFlags)
```
Comment (0xdef1cafe) [inline src/components/Modals/FiatRamps/views/RampsList.tsx:29]: `useFeatureFlag('MtPelerinFiatRamp')` is a lil nicer

**Review by 0xdef1cafe — DISMISSED** [review]
typos and env var plz

File: src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    params.set('sdc', 'EUR')
+  } else {
+    params.set('bdc', mtPeleringSymbol)
+    params.set('bsc', 'EUR')
+  }
+  const network = adapters.getMtPelerinNetFromMtPelerinAssetSymbol(assetId)
+  params.set('net', network)
+  params.set('nets', network)
+  // TODO(stackedq): get the real referral code
+  params.set('rfr', 'shapeshift')
```
[the remark this answers is not in the document]
Comment (stackedq) [reply src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts]: changed to env var

File: src/components/Modals/FiatRamps/config.ts [hunk 3]
```diff
@@ -115,4 +117,27 @@ export const supportedFiatRamps: SupportedFiatRamp = {
...
+  MtPelerin: {
+    label: 'fiatRamps.mtPelerin',
+    tags: ['fiatRamps.noKYC', 'fiatRamps.nonUS'],
+    logo: MtPelerinLogo,
+    isImplemented: true,
+    supportsBuy: true,
+    supportsSell: true,
+    // https://developers.mtpelerin.com/service-information/pricing-and-limits#limits-2
+    // 50 CHS is currently equivalent to 51.72 USD, maybe get the actual rate from redux fiat market data?
+    minimumSellThreshold: 52,
```
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/config.ts:129]: Let's bump this one a bit for safety against CHF<->USD variations?

File: src/components/Modals/FiatRamps/config.ts [hunk 4]
```diff
@@ -115,4 +117,27 @@ export const supportedFiatRamps: SupportedFiatRamp = {
...
   },
+  MtPelerin: {
+    label: 'fiatRamps.mtPelerin',
+    tags: ['fiatRamps.noKYC', 'fiatRamps.nonUS'],
+    logo: MtPelerinLogo,
+    isImplemented: true,
+    supportsBuy: true,
+    supportsSell: true,
+    // https://developers.mtpelerin.com/service-information/pricing-and-limits#limits-2
+    // 50 CHS is currently equivalent to 51.72 USD, maybe get the actual rate from redux fiat market data?
```
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/config.ts:128]:
```suggestion
    // 50 CHF is currently equivalent to 51.72 USD, maybe get the actual rate from redux fiat market data?
```

Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/config.ts:128]: Also might be good to add an additional comment line explaining Mt Pelerin has a minimum of 50 CHF, and our fiat balance is denoted in USD

File: src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+  // TODO: this line and its usages should be removed when Avalanche flag got removed.
+  const avalancheFlag = store.getState().preferences.featureFlags.Avalanche
+
+  const mtPelerinAssets = Object.values(data)
+
+  const assets = mtPelerinAssets.reduce<FiatRampAsset[]>((acc, asset) => {
+    const { symbol } = asset
+    // MtPelerin supports multiple networks for a given asset,
+    // so if the asset is already proccessed, skip to the next one
```
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts:45]: Is this comment still relevant?

File: src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+      ? getConfig().REACT_APP_MTPELERIN_SELL_URL
+      : getConfig().REACT_APP_MTPELERIN_BUY_URL,
+  )
+  const params = new URLSearchParams()
+
+  params.set('type', 'direct-link')
+  params.set('tab', action === FiatRampAction.Sell ? 'sell' : 'buy')
+  params.set('tabs', action === FiatRampAction.Sell ? 'sell' : 'buy')
+  if (action === FiatRampAction.Sell) {
+    params.set('ssc', mtPelerinSymbol)
```
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts:88]: Comment what's this param?

File: src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    params.set('sdc', 'EUR')
+  } else {
+    params.set('bdc', mtPelerinSymbol)
```
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts:91]: Comment what's this param?

File: src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    params.set('bsc', 'EUR')
```
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts:92]: Comment what's this param?

File: src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+      : getConfig().REACT_APP_MTPELERIN_BUY_URL,
+  )
+  const params = new URLSearchParams()
+
+  params.set('type', 'direct-link')
+  params.set('tab', action === FiatRampAction.Sell ? 'sell' : 'buy')
+  params.set('tabs', action === FiatRampAction.Sell ? 'sell' : 'buy')
+  if (action === FiatRampAction.Sell) {
+    params.set('ssc', mtPelerinSymbol)
+    params.set('sdc', 'EUR')
```
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts:89]: Comment what's this param?

File: src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+import { store } from 'state/store'
+
+import { FiatRampAction, FiatRampAsset } from '../FiatRampsCommon'
+
+const moduleLogger = logger.child({
+  namespace: ['Modals', 'FiatRamps', 'fiatRampProviders', 'MtPelerin'],
+})
+
+type MtPelerinResponse = {
+  [k: string]: {
```
Comment (gomesalexandre) [inline src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts:14]: Give a semantic name to the key?

**Review by gomesalexandre — DISMISSED** [review]
Tested with linked lib and LGTM:

- Avalanche assets are properly disabled for Native wallet both with the flag on and off
- With an AVAX-supporting wallet (tested with MM), Avalanche assets are displayed with the flag on, and disabled with the flag on
- Assets available on both Avalanche and Mainnet are displayed, with the `on Avalanche` mention appended to the Avalanche ones
- EUR is used as a default fiat currency on Mt Pelerin link

Added some smallish comments, and this will require a lib bump but stamping it already as this conceptually does what it says on the box

File: src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts (hunk 5 above)
Comment (stackedq) [reply src/components/Modals/FiatRamps/fiatRampProviders/mtpelerin.ts:45]: yup

**Review by gomesalexandre — APPROVED** [review]
Stamping as previously accepted and there are no runtime changes

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
not working in preview environment

**Comment by 0xdef1cafe** [comment]
image

**Comment by stackedq** [comment]
@0xdef1cafe is `REACT_APP_MTPELERIN_ASSETS_API` env var set in fleek?

**Comment by stackedq** [comment]
```
REACT_APP_MTPELERIN_ASSETS_API=https://api.mtpelerin.com/currencies/tokens
REACT_APP_MTPELERIN_BUY_URL=https://buy.mtpelerin.com/
REACT_APP_MTPELERIN_SELL_URL=https://sell.mtpelerin.com/
```

**Review by 0xdef1cafe — APPROVED** [review]
added env vars to all environments

**Pull request merged**